services/web-server/src/servers/createSubscriptionServer.js

```diff
@@ -2,16 +2,48 @@ import { SubscriptionServer } from 'subscriptions-transport-ws';
 import { execute, subscribe } from 'graphql';
 import credentials from './credentials.js';
 import formatError from './formatError.js';
+import scopeUtils from 'taskcluster-lib-scopes';
+import taskcluster from 'taskcluster-client';
 
 export default ({ server, schema, context, path }) =>
   SubscriptionServer.create(
     {
       schema,
       execute,
       subscribe,
-      onConnect(params, socket) {
-        return new Promise(resolve => {
-          credentials()(socket.upgradeReq, {}, resolve);
+      async onConnect(params, socket) {
+
+        return new Promise((resolve, reject) => {
+          credentials()(socket.upgradeReq, {}, async () => {
+            try {
+
+              const token = params.Authorization;
+
+              if (!token) {
+                reject();
+              }
+
+              const credentials= JSON.parse(Buffer.from(token.split(' ')[1], 'base64').toString());
```

Comment:
Check `credentials()` middleware - I think it is already doing some of this - extracts authorization from request and assigns it to `request.credentials`
And then it should likely be reachable at `socket.upgradeReq.credentials`

Reply:
Written the code for decrypting base64 in a separate function.

Reply:
great, thanks!